src/io/flutter/sdk/FlutterSdkManager.java

```diff
@@ -39,6 +42,32 @@ public static FlutterSdkManager getInstance() {
 
   private void listenForSdkChanges() {
     ApplicationLibraryTable.getApplicationTable().addListener(myLibraryTableListener);
+    ProjectManager.getInstance().addProjectManagerListener(new ProjectManagerAdapter() {
+      @Override
+      public void projectOpened(@NotNull Project project) {
+        if (!FlutterSdkUtil.isFluttering()) return;
+        checkForFlutterSdkAddition();
+      }
+      @Override
+      public void projectClosed(@NotNull Project project) {
+        if (FlutterSdkUtil.isFluttering()) return;
+        checkForFlutterSdkRemoval();
+      }
+    });
+  }
+
+  private void checkForFlutterSdkAddition() {
```

Comment:
I wonder if this could take advantage of `FlutterSdkManager` which does a similar check?